app/scripts/modules/core/src/pipeline/manualExecution/manualPipelineExecution.controller.js

```diff
@@ -70,6 +70,13 @@ module.exports = angular.module('spinnaker.core.pipeline.manualPipelineExecution
         return;
       }
 
+      if (trigger && trigger.type === 'manual') {
+        const pipelineTriggerTypes = _.uniq(pipeline.triggers.map(t => t.type));
+        if (pipelineTriggerTypes.length === 1) {
+          trigger.type = pipelineTriggerTypes[0];
+        }
+      }
```

Comment:
if i had to guess i'd say this is doing some normalization for manual re-runs to look right — can you share some context (maybe even a code comment)?

Reply:
I will add a comment to the code